### DISCUSSION TITLE
Installing specified dependencies in build system

### DISCUSSION THREAD

**Discussion by Jomy10** [SwiftPackageIndex/SwiftPackageIndex-Server#3435] [Ideas]
I have a package which relies on ncurses, which fails on Linux bacause it is installed by default.

Would it be a good idea to automatically install packages inside the Dockerfile which are specified by "providers"? For example in my package:
```swift
    .systemLibrary(
        name: "C_ncurses",
        pkgConfig: "ncurses",
        providers: [
            .brew(["ncurses"]),
            .apt(["libncurses5-dev"]),
            .yum(["ncurses-devel"])
        ]),
```
In the dockerfile it could read the Package.swift and run `apt-get install libcurses5-dev`.

I did see the documentation about custom docker images, but it seems a little too much work for small packages.

**Comment by finestructure** [comment]
Thanks for the report, @Jomy10. Bringing ncurses into the base image shouldn't be a problem. I'll take a look!

**Reply by finestructure** [reply]
Could you please tell me the name of your package so I can verify the change?

**Reply by Jomy10** [reply]
> Could you please tell me the name of your package so I can verify the change?

Yes, the package is called SwiftCurses

Here's the link: https://swiftpackageindex.com/Jomy10/SwiftCurses

**Reply by finestructure** [reply]
I've updated the base image and triggered a rebuild on Linux for your package. It's now showing compatibility for the default branch:

![CleanShot 2024-10-17 at 11 04 24@2x](https://github.com/user-attachments/assets/9e139a57-17de-4871-8d6e-91adbb19fde3)

Version 1.0.1 still fails but it looks like an unrelated build error:

```
/host/spi-builder-workspace/Sources/SwiftCurses/Window+attributes.swift:72:24: error: cannot convert value of type 'OpaquePointer' to expected argument type 'UnsafeMutablePointer<WINDOW>?' (aka 'Optional<UnsafeMutablePointer<_win_st>>')
                ncurses.wattron(self.window, attrs.combine())
```

**Reply by Jomy10** [reply]
Thanks a lot! Version 1.0.1 doesn’t support Linux, so this seems right.
